src/Symfony/Component/Translation/IdentityTranslator.php

```diff
@@ -25,17 +25,16 @@ class IdentityTranslator implements LegacyTranslatorInterface, TranslatorInterfa
     use TranslatorTrait;
 
     private $selector;
+    private $debug;
 
     /**
-     * @param MessageSelector|null $selector The message selector for pluralization
+     * @param bool $debug
      */
-    public function __construct(MessageSelector $selector = null)
+    public function __construct(MessageSelector $messageSelector = null, $debug = true)
     {
-        $this->selector = $selector;
-
-        if (__CLASS__ !== \get_class($this)) {
-            @trigger_error(sprintf('Calling "%s()" is deprecated since Symfony 4.2.', __METHOD__), E_USER_DEPRECATED);
-        }
+        @trigger_error(sprintf('Calling "%s()" with a "%s" or "%s" is deprecated since Symfony 4.3.', __METHOD__, MessageSelector::class, null), E_USER_DEPRECATED);
```

Comment:
Triggering the deprecation unconditionally means that there is no way to disable the debug mode unconditionally. We will need to drop the `MessageSelector` type-hint instead:

```php
public function __construct($debug = null)
{
    if (null === $debug || $debug instanceof MessageSelector) {
        @trigger_error(sprintf('Passing a %s instance or null to %s() is deprecated since Symfony 4.4.', MessageSelector::class, __METHOD__), E_USER_DEPRECATED);
        $this->selector = $debug;
        $this->debug = true;
    } elseif (!\is_bool($debug)) {
        throw new \TypeError(sprintf('The first argument of %s() must be a boolean.', __METHOD__));
    } else {
        $this->debug = $debug;
    }
}
```

Reply:
updated